addons/popochiu/engine/objects/character/popochiu_character.gd

```diff
@@ -346,6 +358,25 @@ func play_animation(animation_label: String, animation_fallback := 'idle'):
 	_play_idle()
 
 
+func stop_animation():
+	# TODO: consider returning if idle is playing, but in a way
+	# that's directions and animation_prefix aware!
+	var animation = $AnimationPlayer.get_animation($AnimationPlayer.current_animation)
+	var animation_loop_mode = animation.loop_mode
+	animation.loop_mode = Animation.LOOP_NONE
+	await $AnimationPlayer.animation_finished
+	_play_idle()
+	animation.loop_mode = animation_loop_mode
```

Comment:
What would happen if one wants to stop an animation that is not on loop? Shouldn't this function call $AnimationPlayer.stop()?

Reply:
I was waiting for this question :D

I intended this function to stop the animation organically, when a cycle is over, going back to the idle state. This does not abruptly stop an animation midway through. If a long non-looping animation is running, this function basically does nothing.

Now that I think of it I can just test this case...

Reply:
One more thing is, we can parametrize this function to extend its behavior to stop the animation suddenly, no matter the state.